jsonutil/json.go

```diff
@@ -0,0 +1,40 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package jsonutil
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+// DecodeJsonWithNumbers decodes input data using json.Decoder, ensuring numbers are preserved
```

Comment:
This seems better as _DecodeWithNumber,_ the "Json" part is implied as this is inside jsonutil, and the "Number" here refers to the type name, json.Number, rather than to "numbers" in general.